clang/lib/Sema/SemaTemplateDeduction.cpp

```diff
@@ -493,8 +493,8 @@ DeduceNullPtrTemplateArgument(Sema &S, TemplateParameterList *TemplateParams,
                                                         : CK_NullToPointer)
                     .get();
   return DeduceNonTypeTemplateArgument(
-      S, TemplateParams, NTTP, DeducedTemplateArgument(Value), Value->getType(),
```

Comment:
What is going on with this change?  I don't get what is happening here, can you clarify?

Reply:
Yeah, Value is an expression, but the DeducedTemplateArgument constructor only (implicitly) accepts a TemplateArgument. Instead of `DeducedTemplateArgument(TemplateArgument(Value))`, we go through the implicit path, as it is harmless.

Reply:
Ah!  I see, thanks.  So so we went from:

`DeducedTemplateArgument(TemplateArgument{implicit}(Value))`

to 
`DeducedTemplateArgument{implicit}(TemplateArgument(Value))`

That seems fine to me.  I'd probably prefer it be explicit, but don't feel strongly enough on this one to push back more than this

Reply:
In this case a DeducedTemplateArgument is a TemplateArgument (there is an inheritance relationship).